### DISCUSSION TITLE
How to get data from from initialApolloState on dynamically created pages?

### DISCUSSION THREAD

**Discussion by FRMR1** [vercel/next.js#17885] [Help]
I've reconfigured my app based on [this official example](https://github.com/vercel/next.js/tree/canary/examples/api-routes-apollo-server-and-client) and I've got it to the point where I have my data cached and set to ```initialApolloState```, but I'm having trouble figuring out how to get the extract the data from this object on each page. 

My pages are dynamically created from ```getStaticPaths``` which queries a GraphQL API that is set up to query REST API endpoints. My old setup (which did not allow for SSG, which is why I had to reconfigure), the ```getStaticProps``` function used the page ID parameter to query that specific ID and get all the data needed for that page and set them to props. This new setup however, seems to set the data to the ```initialApolloState``` object with the ID attached to the object name itself. 

For example, one of the prop objects is named ```TourDossier:33887```, so I don't know how I'm supposed to extract any data from this object when it will have a unique name on each page. 

I suppose I'm just lost on how I'm supposed to extract data when using Apollo and SSG using the official example I listed above. The example is very simple and there seems to be another layer of complexity when using ```getStaticPaths``` and the ID parameter from the path in the ```getStaticProps``` query. 

Any help or docs to read would be greatly appreciated.

**Comment by El-Devoper** [comment]
I have the same problem. Have you found a solution?

**Comment by joneslloyd** [comment]
Me too, actually. My initial idea was to grab this from `useRouter()`, but that's only available for client-side stuff (so the `query` property is empty)

**Comment by joneslloyd** [comment]
**UPDATE:**

@FRMR1 @El-Devoper Not sure if this helps/is relevant, but in my situation, I return both the Apollo cache and the required data for my `useQuery` hook from `getStaticProps`, like so:
```
return {
            props: {
                initialApolloState: apolloClient.cache.extract(),
                data: { slug }
            },
        };
```

Then in my component I can grab that `slug` to pass the variable into `useQuery`
